controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java

```diff
@@ -57,10 +78,13 @@
      */
     public void createStream(String scope, String streamName, int minNumSegments, Duration latency) {
         DYNAMIC_LOGGER.incCounterValue(CREATE_STREAM, 1);
-        DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(OPEN_TRANSACTIONS, scope, streamName), 0);
+        // We need to initialize the number of initial segments, but prevent this process to continue reporting the
+        // segment count for this Stream. This will avoid multiple controllers to report conflicting values.
         DYNAMIC_LOGGER.reportGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName), minNumSegments);
-        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_SPLITS, scope, streamName), 0);
-        DYNAMIC_LOGGER.incCounterValue(nameFromStream(SEGMENTS_MERGES, scope, streamName), 0);
+        if (executor != null) {
+            executor.schedule(() -> DYNAMIC_LOGGER.freezeGaugeValue(nameFromStream(SEGMENTS_COUNT, scope, streamName)),
```

Comment:
What happens if this stream never splits or merges (e.g., fixed policy)?

Reply:
In that case this value is not reported after the first time.

Reply:
but what happens to the service relying on this metrics. Does it get stored permanently on something like InfluxDB?

Reply:
Yes, the StatsD reporter can store that metric to a persistent time-series database, such as InfluxDB. As @shiveshr says, the behavioral difference is that before this PR, the Controller handling the `createStream` would be reporting the "initial number of segments" for a given Stream for a period of time (i.e., `dynamicCacheEvictionDurationMs`). With this PR, the Controller handling the `createStream` operation will only report the "initial number of segments" for a Stream only once. With this, we achieve i) to report that information, ii) and do not make that Controller to "smash" the (updated) information being reported by another Controller that handles a `scaleStream` request for that Stream.